edx/frontend-app-course-authoring
# feat: Optionally display message on 403 and support permission hints.

An alert is now displayed when a 403 is returned from the backend
in response to a configuration update.

Additionally, the UI now accepts a `user_permissions` field which
contains hints on what the user can do, so it can improve UX by,
for example, disabling some controls.

Related tickets:
* [TNL-8142](https://openedx.atlassian.net/browse/TNL-8142)
* [BB-4253](https://tasks.opencraft.com/browse/BB-4253)

# Testing

This takes advantage of the changes in edx/edx-platform#27825 but works as previously without them.

1. Launch the service with the demo course;
2. Select a provider with a course or global staff user;
3. Verify:
    1. Global staff user can still change providers;
    2. Provider selection checkboxes are disabled for course staff users.

Also, if a course staff user forcibly re-enables the checkboxes (by using devtools) to force a provider change, an alert is displayed above the form with a description of the specific error (instead of a modal with a generic message).
